### PR TITLE
Throw Error When Using Unsupported Linux ARM

### PR DESCRIPTION
**Pull request by shbenzer**
### **User description**
Added code for Issue #13793 in Selenium Manager

### Description
Selenium Manager Java should now throw an error if using unsupported Linux ARM architecture

### Motivation and Context
Issue #13793 

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

___

### **PR Type**
Bug fix

___

### **Description**
- Added error handling in `SeleniumManager` to throw an exception when an unsupported Linux ARM64 architecture is detected.
- Ensures that users are informed about the lack of support for Linux ARM64 in Selenium Manager.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Error handling</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>SeleniumManager.java</strong><dd><code>Add error handling for unsupported Linux ARM architecture</code></dd></summary>
<hr>

java/src/org/openqa/selenium/manager/SeleniumManager.java

<li>Added a check for Linux ARM architecture.<br> <li> Throws a <code>WebDriverException</code> if ARM64 architecture is detected.<br> <li> Ensures compatibility warning for unsupported architectures.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/14616/files#diff-bc018f8fe13b0ca29c19df281b697583c7ab6ff28e052677871ad22c300464d2">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Comment by shbenzer** [comment]
I think this should be sufficient - but let me know if I misunderstood the decision y'all made in Issue #13793 . Used System.getProperty() instead of current.Is() since ARM is part of the system architecture.

File: java/src/org/openqa/selenium/manager/SeleniumManager.java [hunk 1]
```diff
@@ -194,7 +194,11 @@ private synchronized Path getBinary() {
         } else if (current.is(MAC)) {
           folder = "macos";
         } else if (current.is(LINUX)) {
+          if (System.getProperty("os.arch").contains("arm")) {
+            throw new WebDriverException("Linux ARM is not supported by Selenium Manager");
+          } else {
           folder = "linux";
```
Comment (AutomatedTester) [inline java/src/org/openqa/selenium/manager/SeleniumManager.java]: can we check the formatting of this conditional

Comment (shbenzer) [reply java/src/org/openqa/selenium/manager/SeleniumManager.java]: fixed

**Comment by shbenzer** [comment]
the weird git diff error in the workflow should be fixed now - for some reason the linux word got removed, I put it back

**Review by A1exKH — COMMENTED** [review]
LGTM.

**Comment by shbenzer** [comment]
@pujagani Failing test is unrelated:

> java.lang.Exception: org.openqa.selenium.JavascriptEnabledDriverTest.testShouldBeAbleToFindElementAfterJavascriptCausesANewPageToLoad is not yet expected to work on remote builds using CHROME, but it already works!
	at org.openqa.selenium.testing.SeleniumExtension.afterEach(SeleniumExtension.java:160)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

**Comment by shbenzer** [comment]
looks like the same test is failing @pujagani

> java.lang.Exception: org.openqa.selenium.JavascriptEnabledDriverTest.testShouldBeAbleToFindElementAfterJavascriptCausesANewPageToLoad is not yet expected to work on remote builds using CHROME, but it already works!

Seems like a good thing though?

**Comment by pujagani** [comment]
Yes, the test is not a concern. Thank you @shbenzer!

**Review by pujagani — APPROVED** [review]

File: java/src/org/openqa/selenium/manager/SeleniumManager.java [hunk 2]
```diff
@@ -194,7 +194,11 @@ private synchronized Path getBinary() {
         } else if (current.is(MAC)) {
           folder = "macos";
         } else if (current.is(LINUX)) {
-          folder = "linux";
+          if (System.getProperty("os.arch").contains("arm")) {
```
Comment (mkurz) [inline java/src/org/openqa/selenium/manager/SeleniumManager.java:197]: You should also check for `aarch64`, see 
- #16381

Comment (shbenzer) [reply java/src/org/openqa/selenium/manager/SeleniumManager.java:197]: Create an issue and we’d be happy to take a look - commenting here will cause us to miss or forget @mkurz

Comment (mkurz) [reply java/src/org/openqa/selenium/manager/SeleniumManager.java:197]: I referenced the PR I created. I just wanted to make the author of the PR aware that the check was not enough. (Also for readers who stumple on this issue later)

Comment (shbenzer) [reply java/src/org/openqa/selenium/manager/SeleniumManager.java:197]: Ah, I see - I didn't understand the link clearly on my phone. Thank you!

**Pull request merged**